apps/comps/e2e/utils/rpc-client-helpers.ts

```diff
@@ -80,3 +158,456 @@ export async function createTestRpcClient(
     },
   });
 }
+
+/**
+ * Create a test admin RPC client with optional custom headers
+ *
+ * @param options - Optional configuration
+ * @param options.apiKey - API key to include in Authorization header
+ * @param options.headers - Custom headers to include
+ */
+export async function createTestAdminRpcClient(options?: {
+  apiKey?: string;
+  headers?: Record<string, string>;
+}): Promise<RouterClient<typeof adminRouter>> {
+  const headers = { ...options?.headers };
+
+  // Add Authorization header if apiKey is provided
+  if (options?.apiKey) {
+    headers.authorization = `Bearer ${options.apiKey}`;
+  }
+
+  return createRouterClient(adminRouter, {
+    context: {
+      cookies: createMockCookies("dummy-token"),
+      headers: createMockHeaders(headers),
+      params: createMockParams(),
+      adminService,
+      boostBonusService,
+      userService,
+      competitionService,
+      competitionRepository,
+      agentService,
+      arenaService,
+      partnerService,
+      balanceService,
+      portfolioSnapshotterService,
+      rewardsService,
+      logger: createLogger("TestAdminRpcClient"),
+    },
+  });
+}
+
+/**
+ * Assert that a promise throws an error with a specific error code
+ * Useful for testing RPC error responses
+ */
+export async function assertRpcError<TOutput>(
+  promise: Promise<TOutput>,
+  errorCode: string,
+  options?: {
+    messageContains?: string | string[];
+  },
+): Promise<void> {
+  const result = expect(promise).rejects;
+  await result.toThrow();
+  await result.toMatchObject({
+    code: errorCode,
+  });
+  if (options?.messageContains) {
+    const messageContainsArray = Array.isArray(options.messageContains)
+      ? options.messageContains
+      : [options.messageContains];
+
+    for (const messageContains of messageContainsArray) {
```

Comment:
this seems overly complicated?  Most tests just use something like `await expect(...).toThrow(/<err messsage>/)`